### PR TITLE
Fix Team URL aliasing issues

### PR DESCRIPTION
**Pull request by bkDJ**
Team URLs now include challenge name
* moved around files so they make sense with new paths
* old URLs will redirect to first team with same name
* updated all links (e.g. emails) and redirects (e.g. create/edit actions or permissions failures)
* updated tests
* updated ActiveAdmin (can see teams of specific challenges, or all teams, fixed URLs, etc)

### CODE REVIEW

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# This is a legacy controller.
+# A team without a specified challenge might have name aliasing issues
+# so we redirect to the correct page
+class TeamsController < ApplicationController
```
Comment (skbly7) [inline app/controllers/teams_controller.rb]: Does this be here?

```
before_action :authenticate_participant!, except: [:show]
```

**Comment by nemaniarjun** [comment]
Build is failing

File: app/controllers/teams_controller.rb (hunk 1 above)
Comment (bkDJ) [reply app/controllers/teams_controller.rb]: The only action in this controller is `show`. So a `before_action` callback that would only happen for non-`show` actions is unnecessary.

Comment (nemaniarjun) [reply app/controllers/teams_controller.rb]: @skbly7 also the authenticate_participant callback is used to make sure that a user is logged in, we dont need that here.

**Pull request merged**